### PR TITLE
feat:Added user information for commit validator

### PR DESCRIPTION
**Pull request by fengwei-tian**
1. Added the logic parsing the listCommits response: now it will also have author/committer information in the response
2. Added a new option in the messageSettings: messageType(author_email/committer_email) 
3. Added the logic to filter commit author/committer email

### CODE REVIEW

**Comment by jusx** [comment]
@fengwei-tian Looks good, but could you also update the docs with these new options?

**Review by fengwei-tian — COMMENTED** [review]
Maybe add - Only check the regex against the xxx

**Comment by eagraf** [comment]
@jusx just added some documentation for the `message_type` option

File: lib/validators/commit.js [hunk 1]
```diff
@@ -73,14 +77,22 @@ class Commit extends Validator {
       orderedCommits = [orderedCommits[orderedCommits.length - 1]]
     }
 
-    const commitMessages = orderedCommits.map(commit => commit.message)
+    let commitMessages = orderedCommits.map(commit => commit.message)
+    let FAIL_MESSAGE = DEFAULT_FAIL_MESSAGE
```
Comment (jusx) [inline lib/validators/commit.js:81]: Since this isn't a const, I suggest changing this to `failMessage` as the variable name.

**Review by jusx — CHANGES_REQUESTED** [review]

**Comment by eagraf** [comment]
Any updates @jusx, thanks!

**Comment by ankurd1** [comment]
Hey @jusx were you able to review this? We'd like to get this merged as it's blocking mergable adoption for us. Thanks!

**Comment by RemyDeWolf** [comment]
Hi @jusx , we have addressed your comments?
Might taking another look? 
I will personally buy coffee if you have time for us :) https://www.buymeacoffee.com/mergeable

**Review by shine2lay — APPROVED** [review]
LGTM

**Comment by shine2lay** [comment]
I'll merge it once CI is passing

**Comment by RemyDeWolf** [comment]
@shine2lay Thank you so much for the merge and the quick release.
We are now setting up mergeable at Rippling

**Comment by eagraf** [comment]
@shine2lay Would it be possible to deploy release 2.17? That will help us get up and running at Rippling. Thanks!

**Comment by shine2lay** [comment]
@eagraf our deploy pipeline was broken, I just fixed it, new version should be deployed

**Pull request merged**